packages/ember-glimmer/tests/integration/syntax/if-unless-test.js

```diff
@@ -1,82 +1,27 @@
 import { moduleFor } from '../../utils/test-case';
 import { set } from 'ember-metal/property_set';
-import {
-  BASIC_TRUTHY_TESTS,
-  BASIC_FALSY_TESTS,
-  SharedSyntaxConditionalsTest
-} from '../../utils/shared-conditional-tests';
+import { TogglingSyntaxConditionalsTest } from '../../utils/shared-conditional-tests';
 
-moduleFor('Syntax test: {{#if}}', class extends SharedSyntaxConditionalsTest {
+moduleFor('Syntax test: {{#if}} with inverse', class extends TogglingSyntaxConditionalsTest {
 
   templateFor({ cond, truthy, falsy }) {
     return `{{#if ${cond}}}${truthy}{{else}}${falsy}{{/if}}`;
   }
 
-  ['@test it renders and hides the given block based on the conditional']() {
-    this.render(`{{#if cond1}}T1{{/if}}{{#if cond2}}T2{{/if}}`, { cond1: true, cond2: false });
+});
 
-    this.assertText('T1');
-
-    this.runTask(() => this.rerender());
-
-    this.assertText('T1');
-
-    this.runTask(() => set(this.context, 'cond1', false));
-
-    this.assertText('');
-
-    this.runTask(() => {
-      set(this.context, 'cond1', true);
-      set(this.context, 'cond2', true);
-    });
-
-    this.assertText('T1T2');
-
-    this.runTask(() => {
-      set(this.context, 'cond1', true);
-      set(this.context, 'cond2', false);
-    });
-
-    this.assertText('T1');
-  }
-
-}, BASIC_TRUTHY_TESTS, BASIC_FALSY_TESTS);
-
-moduleFor('Syntax test: {{#unless}}', class extends SharedSyntaxConditionalsTest {
+moduleFor('Syntax test: {{#unless}} with inverse', class extends TogglingSyntaxConditionalsTest {
 
   templateFor({ cond, truthy, falsy }) {
     return `{{#unless ${cond}}}${falsy}{{else}}${truthy}{{/unless}}`;
   }
 
-  ['@test it renders and hides the given block based on the conditional']() {
-    this.render(`{{#unless cond1}}F1{{/unless}}{{#unless cond2}}F2{{/unless}}`, {
-      cond1: true,
-      cond2: false
-    });
-
-    this.assertText('F2');
-
-    this.runTask(() => this.rerender());
-
-    this.assertText('F2');
-
-    this.runTask(() => set(this.context, 'cond2', true));
+});
 
-    this.assertText('');
+moduleFor('Syntax test: {{#if}} and {{#unless}} without inverse', class extends TogglingSyntaxConditionalsTest {
 
-    this.runTask(() => {
-      set(this.context, 'cond1', false);
-      set(this.context, 'cond2', false);
-    });
-
-    this.assertText('F1F2');
-
-    this.runTask(() => {
-      set(this.context, 'cond1', true);
-      set(this.context, 'cond2', false);
-    });
-
-    this.assertText('F2');
+  templateFor({ cond, truthy, falsy }) {
+    return `{{#if ${cond}}}${truthy}{{/if}}{{#unless ${cond}}}${falsy}{{/unless}}`;
   }
```

Comment:
@GavinJoyce I finally cracked the nut on how to test "without inverse" as part of the bigger test suite! The trick is to generate the if/unless as a pair, both without inverses. I think you can probably use this trick for the helpers too once you have implemented inline-unless.